### ISSUE TITLE
stm32f4:rcc: flash config set twice

### ISSUE THREAD

**Issue by alexkazik** [libopencm3/libopencm3#1014]
`clock->flash_config` does contain the wait state and cache config.
The caches are set.
And `flash_set_ws` clears the wait state bits and then or's the flash config and thus setting the cache again.

https://github.com/libopencm3/libopencm3/blob/1adc418f9afaa5b99c34f3eea797dc4c1155c056/lib/stm32/f4/rcc.c#L798

**Comment by karlp** [comment]
yep. should be separated. feel free to send a pr,  but try not to break existing users too much.

**Comment by wingunder** [comment]
@alexkazik, I hope you don't mind, but I took the liberty to submit a fix for this problem: #1209

**Comment by alexkazik** [comment]
I don't mind. I forgot about this because I write in assembler now (at least for most parts).
